quint/test/runtime/compile.test.ts

```diff
@@ -80,56 +80,86 @@ function assertVarExists(kind: ComputableKind, name: string, input: string) {
   res.mapLeft(m => assert.fail(m))
 }
 
+// compile a computable for a run definition
+function callableFromContext(ctx: CompilationContext, callee: string): Either<string, Callable> {
+  let key = undefined
+  const lastModule = ctx.compilationState.modules[ctx.compilationState.modules.length - 1]
+  const def = lastModule.declarations.find(def => def.kind === 'def' && def.name === callee)
+  if (!def) {
+    return left(`${callee} definition not found`)
+  }
+  key = kindName('callable', def.id)
+  if (!key) {
+    return left(`${callee} not found`)
+  }
+  const run = ctx.evaluationState.context.get(key) as Callable
+  if (!run) {
+    return left(`${callee} not found via ${key}`)
+  }
+
+  return right(run)
+}
+
 // Scan the context for a callable. If found, evaluate it and return the value of the given var.
-// Assumes the input has a single callable
-// `callee` is used for error reporting.
-function evalVarAfterCall(varName: string, callee: string, input: string): Either<string, string> {
+// Assumes the input has a single callable `callee` is used for error reporting.
+function evalVarAfterRun(varName: string, callee: string, input: string): Either<string, string> {
   // use a combination of Maybe and Either.
   // Recall that left(...) is used for errors,
   // whereas right(...) is used for non-errors in sweet monads.
   const callback = (ctx: CompilationContext): Either<string, string> => {
-    let key = undefined
-    const lastModule = ctx.compilationState.modules[ctx.compilationState.modules.length - 1]
-    const def = lastModule.declarations.find(def => def.kind === 'def' && def.name === callee)
-    if (!def) {
-      return left(`${callee} definition not found`)
-    }
-    key = kindName('callable', def.id)
-    if (!key) {
-      return left(`${callee} not found`)
-    }
-    const run = ctx.evaluationState.context.get(key)
-    if (!run) {
-      return left(`${callee} not found via ${key}`)
-    }
+    return callableFromContext(ctx, callee)
+      .mapRight(run => {
+        return run
+          .eval()
+          .map(res => {
+            if ((res as RuntimeValue).toBool() === true) {
+              // extract the value of the state variable
+              const nextVal = (ctx.evaluationState.context.get(kindName('nextvar', varName)) ?? fail).eval()
+              // using if-else, as map-or-unwrap confuses the compiler a lot
```

Comment:
I don't know what to do with this comment.
Can we make it more relevant for the reader?